services/121-service/src/payments/fsp-integration/intersolve-visa/intersolve-visa.api.service.ts

```diff
@@ -66,50 +91,131 @@ export class IntersolveVisaApiService {
     return timeLeftBeforeExpire > 60000;
   }
 
-  public async createCustomer(
-    payload: IntersolveCreateCustomerDto,
-  ): Promise<IntersolveCreateCustomerResponseBodyDto> {
+  public async createCustomer(input: {
+    externalReference: string;
+    name: string;
+    contactInformation: ContactInformation;
+    estimatedAnnualPaymentVolumeMajorUnit: number;
+  }): Promise<CreateCustomerReturnType> {
+    // Create the request body to send
+    const createCustomerRequestDto: CreateCustomerRequestDto = {
+      externalReference: input.externalReference, // The IntersolveVisa does not "know about this", but we pass in the registration.referenceId here.
```

Comment:
For me personally this shorthand forces me to think and interpret what it means, whereas as it is now, makes it very explicit and easy to read for me: the externalReference parameter that this function expects, is assigned the externalReference value that was received as input of this function. But maybe that is getting used to TS as language?

Reply:
> But maybe that is getting used to TS as language?

Yes indeed, because for me it is the opposite 😄 

And FWIW, this is JS, not TS, and more specifically the [shorthand property feature of ES6](https://ui.dev/shorthand-properties#shorthand-properties) (which is a well known JS release, because it is the release that "modernized the language" introducing also other things like `const` & `let`, arrow functions, etc.)

Reply:
Discussed => apply together with changing function params. 

We can probably enforce this with eslint. @aberonni  will look at this.

Reply:
Will be addressed in AB#30233